rust/src/applayer.rs

```diff
@@ -445,7 +445,7 @@ pub type StateGetTxFn            = unsafe extern "C" fn (*mut c_void, u64) -> *m
 pub type StateGetTxCntFn         = unsafe extern "C" fn (*mut c_void) -> u64;
 pub type StateGetProgressFn = unsafe extern "C" fn (*mut c_void, u8) -> c_int;
 pub type GetEventInfoFn     = unsafe extern "C" fn (*const c_char, *mut c_int, *mut AppLayerEventType) -> c_int;
-pub type GetEventInfoByIdFn = unsafe extern "C" fn (c_int, *mut *const c_char, *mut AppLayerEventType) -> i8;
+pub type GetEventInfoByIdFn = unsafe extern "C" fn (c_int, *mut *const c_char, *mut AppLayerEventType) -> c_int;
```

Comment:
That that we also use `uint8_t` and `int` for the event_type (first argument) in C, while not breaking anything right now, could be a source for error.